lib/elastomer/client/bulk.rb

```diff
@@ -43,6 +43,48 @@ def bulk( body = nil, params = nil )
       end
     end
 
+    # Stream bulk actions from an Enumerator.
+    #
+    # Examples
+    #
+    #   ops = [
+    #     Bulk::index(document1, :_type => 'default-type', :_id => 1),
+    #     Bulk::index(document2, params2),
+    #     Bulk::delete(params3),
+    #   ]
```

Comment:
I'm not a huge fan of the `Bulk::index` / `Bulk::delete` syntax here. My proposal is this:

```ruby
ops = [
  [:index, document1, {:_type => "foo", :_id => 1}],
  [:create, document2],
  [:delete, {:_type => "bar", :_id => 42}]
]
```

The first item in the list is the bulk action to take. This corresponds to one of the action methods on the Bulk instance. The second item in the list is a document (or the params in the case of `:delete`). And thirdly comes the optional params.

Inside the Enumerator we have:

```ruby
Enumerator.new do |yielder|
  ops.each do |action, *args|
    response = bulk_obj.send(action, *args)
    yielder.yield response unless response.nil?
  end
  response = bulk_obj.call
  yielder.yield response unless response.nil?
end
```

The Ruby `send` method is being used here to call the `action` method on the `bulk_obj` and passing the `*args` to the method.